src/components/TodoItem.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import classNames from 'classnames';
+import {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import { DispatchContext, StateContext } from '../Store';
+import { Todo } from '../utils/interfaces';
+import { ActionType } from '../utils/enums';
+
+type Props = {
```

Comment:
Use interface prefer.
Type has unique ability but if you don`t need them use interface. It is sampled in scaled and has more interesting features.